engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -742,43 +742,47 @@ struct unfurling_darkness_t final : public priest_spell_t
 // ==========================================================================
 // Vampiric Touch
 // ==========================================================================
-struct vampiric_touch_heal_t final : public priest_heal_t
+struct vampiric_touch_t final : public priest_spell_t
 {
-  vampiric_touch_heal_t( priest_t& p ) : priest_heal_t( "vampiric_touch_heal", p, p.dot_spells.vampiric_touch )
+  struct vampiric_touch_heal_t final : public priest_heal_t
   {
-    background         = true;
-    may_crit           = false;
-    may_miss           = false;
-    base_dd_multiplier = 1.0;
+    vampiric_touch_heal_t( priest_t& p ) : priest_heal_t( "vampiric_touch_heal", p, p.dot_spells.vampiric_touch )
+    {
+      background         = true;
+      may_crit           = false;
+      may_miss           = false;
+      base_dd_multiplier = 1.0;
 
-    // Turn off Insanity gen from hit action
-    energize_type     = action_energize::NONE;  // no insanity gain
-    energize_amount   = 0;
-    energize_resource = RESOURCE_NONE;
+      // Turn off Insanity gen from hit action
+      energize_type     = action_energize::NONE;  // no insanity gain
+      energize_amount   = 0;
+      energize_resource = RESOURCE_NONE;
 
-    // Turn off all damage parts of the spell
-    spell_power_mod.direct = spell_power_mod.tick = base_td_multiplier = 0;
-    dot_duration                                                       = timespan_t::from_seconds( 0 );
+      // Turn off all damage parts of the spell
+      spell_power_mod.direct = spell_power_mod.tick = base_td_multiplier = 0;
+      dot_duration                                                       = timespan_t::from_seconds( 0 );
 
-    // TODO: Confirm if this healing can proc trinkets/etc
-    callbacks = false;
-  }
+      // TODO: Confirm if this healing can proc trinkets/etc
+      callbacks = false;
+    }
 
-  void trigger( double original_amount )
-  {
-    base_dd_min = base_dd_max = original_amount * data().effectN( 2 ).m_value();
-    execute();
-  }
-};
+    void trigger( double original_amount )
+    {
+      base_dd_min = base_dd_max = original_amount * data().effectN( 2 ).m_value();
+      execute();
+    }
+  };
 
-struct vampiric_touch_t final : public priest_spell_t
-{
+  vampiric_touch_heal_t* vampiric_touch_heal;
```

Comment:
you could add the propagate_const wrapper here as well, and just initialize the heal in the initializer list instead of in the constructor body.
Same for DP.